pkg/security/securitycontextconstraints/selinux/mustrunas.go

```diff
@@ -67,3 +71,70 @@ func (s *mustRunAs) Validate(pod *api.Pod, container *api.Container) field.Error
 
 	return allErrs
 }
+
+func equalLevels(expectedLevel, actualLevel string) bool {
+	if expectedLevel == actualLevel {
+		return true
+	}
+
+	// "s0:c6,c0" => [ "s0", "c6,c0" ]
+	expectedParts := strings.SplitN(expectedLevel, ":", 2)
+	actualParts := strings.SplitN(actualLevel, ":", 2)
+	if len(expectedParts) != 2 || len(expectedParts) != len(actualParts) {
+		return false
+	}
+
+	// "s0-s0" => [ "s0" ]
+	expectedSensitivity := parseSensitivity(expectedParts[0])
+	actualSensitivity := parseSensitivity(actualParts[0])
+	if !reflect.DeepEqual(expectedSensitivity, actualSensitivity) {
+		return false
+	}
+
+	// "c6,c0" => [ "c0", "c6" ]
+	expectedCategories := parseCategories(expectedParts[1])
+	actualCategories := parseCategories(actualParts[1])
+
+	return reflect.DeepEqual(expectedCategories, actualCategories)
+}
+
+func parseSensitivity(sensitivity string) []string {
+	// "s0-s0" => [ "s0" ]
+	if strings.IndexByte(sensitivity, '-') > -1 {
+		sensitivityRange := strings.SplitN(sensitivity, "-", 2)
+		if sensitivityRange[0] == sensitivityRange[1] {
+			return []string{sensitivityRange[0]}
+		}
+	}
+
+	return []string{sensitivity}
+}
+
+func parseCategories(categories string) []string {
+	parts := strings.Split(categories, ",")
+
+	// "c0.c3" => [ "c0", "c1", "c2", c3" ]
+	if len(parts) == 1 && strings.IndexByte(categories, '.') > -1 {
+		categoryRange := strings.SplitN(categories, ".", 2)
+		if len(categoryRange) == 2 && categoryRange[0][0] == 'c' && categoryRange[1][0] == 'c' {
+			begin := strings.TrimPrefix(categoryRange[0], "c")
+			end := strings.TrimPrefix(categoryRange[1], "c")
+
+			// bitSize 16 because we expect that categories will be in a range [0, 1024)
+			from, err1 := strconv.ParseInt(begin, 10, 16)
+			to, err2 := strconv.ParseInt(end, 10, 16)
+			if err1 == nil && err2 == nil && from < to {
+				parts = make([]string, to-from+1)
+				for i := from; i <= to; i++ {
+					parts[i] = fmt.Sprintf("c%d", i)
```

Comment:
This will go out of bounds if `categoryRange[0]` is not `c0` (e.g., `parseCategories("c1.c4")` panics—might be a good test case).  You cannot use the same variable for the index and the format, at least not without some arithmetic.

Reply:
@Miciah Indeed, good catch, thanks!